src/main/java/io/sniffy/socket/SnifferSocketImpl.java

```diff
@@ -0,0 +1,375 @@
+package io.sniffy.socket;
+
+import io.sniffy.Sniffer;
+import io.sniffy.util.StackTraceExtractor;
+import io.sniffy.util.ExceptionUtil;
+
+import java.io.FileDescriptor;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+class SnifferSocketImpl extends SocketImpl {
+
+    private final SocketImpl delegate;
+
+    private InetSocketAddress address;
+
+    private final static AtomicInteger counter = new AtomicInteger();
+
+    private final int id = counter.getAndIncrement();
+
+    SnifferSocketImpl(SocketImpl delegate) {
+        this.delegate = delegate;
+    }
+
+    void logSocket(long millis) {
+        logSocket(millis, 0, 0);
+    }
+
+    void logSocket(long millis, int bytesDown, int bytesUp) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/sniffy/sniffy/file/2495582801/issues/source?bid=3005960&fileBranchId=3034483#l34)